client/state/data-layer/wpcom/theme-filters/index.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * Internal dependencies
+ */
+import { THEME_FILTERS_REQUEST } from 'state/action-types';
+import {
+	receiveThemeFilters,
+	receiveThemeFiltersFailure,
+} from 'state/themes/actions';
+import wpcom from 'lib/wp';
+
+export const fetchThemeFilters = ( { dispatch } ) =>
+	wpcom.req.get( '/theme-filters', { apiVersion: '1.2' } )
+		.then( ( themeFilters ) => {
+			dispatch( receiveThemeFilters( themeFilters ) );
+		} )
+		.catch( ( error ) => {
+			dispatch( receiveThemeFiltersFailure( error ) );
+		} );
```

Comment:
Check out [the HTTP layer](https://github.com/automattic/wp-calypso/tree/master/client/state/data-layer/wpcom-http) for these kinds of requests. We can dramatically simplify our tests (because the HTTP layer already has its own unit tests) and we can remove some statefulness from the process.

```js
const fetchFilters = ( { dispatch }, action ) => dispatch( http( {
	method: 'GET',
	apiVersion: '1.2',
	path: '/theme-filters',
}, action ) );

const storeFilters = ( { dispatch }, action, next, data ) =>
	validate( data )
		? dispatch( {
			type: THEME_FILTERS_ADD,
			filters: fromApi( data.filters )
		  } )
		: dispatch( createNotice( 'Wrong API response format' ) );

const reportError = ( { dispatch } ) => dispatch( createNotice( 'Something bad happened' )

export default {
	[ THEME_FILTERS_REQUEST ]: dispatchRequest( fetchFilters, storeFilters, reportError )
}
```

this ^^^ is just to illustrate the idea and not supposed to represent shippable code